src/thing_creature.c

```diff
@@ -1803,7 +1803,7 @@ void creature_cast_spell(struct Thing *castng, SpellKind spl_idx, long shot_lvl,
         cctrl->teleport_y = trg_y;
     }
 
-    if (spconf->caster_affected && castng->index == cctrl->targtng_idx)
+    if (spconf->caster_affected)
```

Comment:
It is a design choice we have to make. We need to set rules for setting the target parameter of the instance, i.e. the 3rd parameter of set_creature_instance().

I suggest that for teleport we set caster itself as the target parameter. It can imply that the spell teleport "caster itself" to somewhere. In the future, we could extend its function to allow it to teleport "another creature" to somewhere.